### DISCUSSION TITLE
Ability to select a name for project.json files

### DISCUSSION THREAD

**Discussion by khludenevav** [nrwl/nx#27299] [Feature Requests]
I'm inside a big monorepo. And currently my nx.json is placed in `<repoRoot>/client`.
I'm want to move nx.json to repo root and nx stop working because there are about 50 files in monorepo which have name `project.json`. Nx can't parse them and fails.
The idea is to somehow restrict glob how Nx searches project.json files, or add ability to change these file names for example to `nx_project.json`.
I believe it also speedup project parse, because now  `nx graph` after `nx reset` executes about 20 seconds.

I can't use .nxignore because some of these files are inputs for targets.

Probably we can start to add options to change glob in `getDefaultPlugins` (packages/nx/src/project-graph/plugins/internal-api.ts)
plugin `../../plugins/project-json/build-nodes/project-json`.

**Comment by khludenevav** [comment]
@FrozenPandaz (mentioned as last editor for project-json plugin https://github.com/nrwl/nx/pull/26952)
I wanted to get input for this request. Maybe it goes against nx direction at all. 
For me seems it can be relatively simple enough to implement, probably as the first step - through environment variable.

**Comment by moatorres** [comment]
You could move your NX-related `project.json` content inside `package.json` files by adding `nx` as a property. I'm not really fan of it as it can bloat `package.json` files depending of the complexity of the configurations, but it could serve as a workaround for the time being.

I'm currently using `project.json` for packages/libs but not for applications:

```json
{
  "name": "blog",
  "version": "0.0.1",
  "dependencies": {
    "next": "^15.1.6",
    "react": "^19.0.0",
    "react-dom": "^19.0.0"
  },
  "devDependencies": {
    "@moatorres/ui": "workspace:*"
  },
  "nx": {
    "name": "blog",
    "projectType": "application",
    "sourceRoot": "apps/blog"
  },
  "private": true
}
```

Here's a less contrived example for a package/lib:

```json
{
  "name": "@moatorres/ui",
  "version": "0.0.1",
  "type": "module",
  "main": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "module": "./dist/index.js",
  "exports": {
    "./package.json": "./package.json",
    ".": {
      "types": "./dist/index.d.ts",
      "import": "./dist/index.js",
      "default": "./dist/index.js"
    }
  },
  "dependencies": {
    "@radix-ui/react-accordion": "^1.2.2",
    "clsx": "^2.1.1",
    "tailwind-merge": "^3.0.1"
  },
  "nx": {
    "targets": {
      "build": {
        "executor": "@nx/vite:build",
        "outputs": [
          "{options.outputPath}"
        ],
        "defaultConfiguration": "production",
        "options": {
          "outputPath": "packages/ui/dist"
        },
        "configurations": {
          "development": {
            "mode": "development"
          },
          "production": {
            "mode": "production"
          }
        }
      }
    }
  },
  "peerDependencies": {
    "@types/react": "^19.0.8",
    "@types/react-dom": "^19.0.3",
    "react": "^19.0.0",
    "react-dom": "^19.0.0"
  },
  "peerDependenciesMeta": {
    "@types/react": {
      "optional": true
    },
    "@types/react-dom": {
      "optional": true
    }
  }
}
```

**Reply by khludenevav** [reply]
Yes, seems I have to do that, but ideally I still want nx to avoid scan 95% of repository to build project graph.

UPD: it will not help. Nx still tries to read project.json files from the rest of repository (which is not in NX format) and fail
